gpMgmt/bin/go-tools/hub/make_cluster.go

```diff
@@ -0,0 +1,457 @@
+		} else {
+			hostSegmentMap[seg.HostName] = append(hostSegmentMap[seg.HostName], segReq)
+		}
+	}
+
+	progressLabel := "Initializing primary segments:"
+	progressTotal := len(segs)
+	stream.StreamProgressMsg(progressLabel, progressTotal)
+
+	request := func(conn *Connection) error {
+		var wg sync.WaitGroup
+
+		segs := hostSegmentMap[conn.Hostname]
+		errs := make(chan error, len(segs))
+		for _, seg := range segs {
+			seg := seg
+			wg.Add(1)
+			go func(seg *idl.Segment) {
+				defer wg.Done()
+
+				gplog.Debug(fmt.Sprintf("Starting to create primary segment: %s", seg))
+				err := CreateSingleSegment(conn, seg, clusterParams, coordinatorAddrs)
+				if err != nil {
+					errs <- err
+				} else {
+					stream.StreamProgressMsg(progressLabel, progressTotal)
+					gplog.Debug(fmt.Sprintf("Successfully created primary segment: %s", seg))
+				}
+			}(seg)
+		}
+
+		wg.Wait()
+		close(errs)
+
+		var err error
+		for e := range errs {
+			err = errors.Join(err, e)
+		}
+		return err
+	}
+
+	return ExecuteRPC(s.Conns, request)
+}
+
+func ExecOnDatabase(conn *dbconn.DBConn, dbname string, query string) error {
+	conn.DBName = dbname
+	if err := conn.Connect(1); err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	if _, err := conn.Exec(query); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CreateGpToolkitExt(conn *dbconn.DBConn) error {
+	createExtensionQuery := "CREATE EXTENSION gp_toolkit"
+
+	for _, dbname := range []string{"template1", "postgres"} {
+		if err := execOnDatabaseFunc(conn, dbname, createExtensionQuery); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func ImportCollation(conn *dbconn.DBConn) error {
+	importCollationQuery := "SELECT pg_import_system_collations('pg_catalog'); ANALYZE;"
+
+	if err := execOnDatabaseFunc(conn, "postgres", "ALTER DATABASE template0 ALLOW_CONNECTIONS on"); err != nil {
+		return err
+	}
+
+	if err := execOnDatabaseFunc(conn, "template0", importCollationQuery); err != nil {
+		return err
+	}
+	if err := execOnDatabaseFunc(conn, "template0", "VACUUM FREEZE"); err != nil {
+		return err
+	}
+
+	if err := execOnDatabaseFunc(conn, "postgres", "ALTER DATABASE template0 ALLOW_CONNECTIONS off"); err != nil {
+		return err
+	}
+
+	for _, dbname := range []string{"template1", "postgres"} {
+		if err := execOnDatabaseFunc(conn, dbname, importCollationQuery); err != nil {
+			return err
+		}
+
+		if err := execOnDatabaseFunc(conn, dbname, "VACUUM FREEZE"); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func CreateDatabase(conn *dbconn.DBConn, dbname string) error {
+	createDbQuery := fmt.Sprintf("CREATE DATABASE %s", dbname)
+	if err := execOnDatabaseFunc(conn, "template1", createDbQuery); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func SetGpUserPasswd(conn *dbconn.DBConn, passwd string) error {
+	user, err := utils.System.CurrentUser()
+	if err != nil {
+		return err
+	}
+
+	alterPasswdQuery := fmt.Sprintf("ALTER USER %s WITH PASSWORD '%s'", user.Username, passwd)
```

Comment:
Add quotes to user string, can throw an error when there's a hyphen in the user-name.

```suggestion
	alterPasswdQuery := fmt.Sprintf("ALTER USER \"%s\" WITH PASSWORD '%s'", user.Username, passwd)
```